packages/antd/src/routing/NavTabBarAnt.tsx

```diff
@@ -0,0 +1,119 @@
+import {
+    LocationDependentStateSpaceHandler,
+    LocationDependentStateSpaceHandlerImpl,
+    LocationDependentStateSpaceHandlerProps,
+    SubStateInContext,
+} from '@moxb/moxb';
+import { Tabs } from 'antd';
+import { inject, observer } from 'mobx-react';
+import * as React from 'react';
+import { UIFragment, UIFragmentSpec } from '../not-antd';
+import * as Anchor from '../not-antd/Anchor';
+import { renderSubStateCore } from '../not-antd/rendering';
+
+const TabPane = Tabs.TabPane;
+
+export interface NavTabProps<DataType>
+    extends LocationDependentStateSpaceHandlerProps<UIFragment, UIFragmentSpec, DataType> {
+    /**
+     * Any extra menu items to add
+     */
+    extras?: JSX.Element[];
+
+    /**
+     * Any direct styles to apply
+     */
+    style?: React.CSSProperties;
+
+    /**
+     * Tab bar alignment mode
+     */
+    mode?: 'top' | 'right' | 'bottom' | 'left';
+}
+
+@inject('locationManager')
+@observer
+/**
+ * This widget show an Ant tab bar, based on the state-space.
+ */
+export class NavTabBarAnt<DataType> extends React.Component<NavTabProps<DataType>> {
+    protected getLocationDependantStateSpaceHandler() {
+        const { id, children: _children, extras, style, mode, ...stateProps } = this.props;
+
+        return new LocationDependentStateSpaceHandlerImpl({
+            ...stateProps,
+            id: 'tab bar of ' + (this.props.id || 'no-id'),
+            intercept: true,
+        });
+    }
+
+    // tslint:disable-next-line:cyclomatic-complexity
+    protected renderStateTabPane(
+        states: LocationDependentStateSpaceHandler<UIFragment, UIFragmentSpec, DataType>,
+        state: SubStateInContext<UIFragment, UIFragmentSpec, DataType>
+    ) {
+        const { label, key, menuKey, itemClassName, newWindow, linkStyle, linkClassName, title } = state;
+
+        const url = states.getUrlForSubState(state);
+        const anchorProps: Anchor.UIProps = {
+            label: label || key,
+            href: url,
+            target: newWindow ? '_blank' : undefined,
+            onClick: newWindow ? undefined : () => states.trySelectSubState(state),
+            style: linkStyle,
+            title,
+        };
+        if (linkClassName) {
+            anchorProps.className = linkClassName;
+        }
+        const itemProps: any = {};
+        if (itemClassName) {
+            itemProps.className = itemClassName;
+        }
+        const tabLabel = <Anchor.Anchor {...anchorProps} />;
+        return (
+            <TabPane key={menuKey} tab={tabLabel} {...itemProps}>
```

Comment:
Notice that here, we are using the `menuKey`, instead of the `key`. 
The `menuKey` is automatically generated, and is guaranteed to exist even for the root state. (And also supports multi-level menus.)